lib/mocha.js

```diff
@@ -424,22 +448,42 @@ Mocha.prototype.run = function(fn){
   var options = this.options;
   options.files = this.files;
   var runner = new exports.Runner(suite, options.delay);
-  var reporter = new this._reporter(runner, options);
+  // For each loaded reporter constructor, create
+  // an instance and initialize it with the runner
+  var reporters = this._reporters.map(function (reporterConfig) {
+    var _reporter = reporterConfig[0]
+      , path = reporterConfig[1];
+
+    return new _reporter(runner, options, path);
```

Comment:
In following commits you will be passing a `reporterOptions` object instead of mocha's `options`, I suppose?